coalib/results/result_actions/OpenEditorAction.py

```diff
@@ -39,23 +99,44 @@ def apply(self, result, original_file_dict, file_diff_dict, editor: str):
 
         :param editor: The editor to open the file with.
         """
-        # Use set to remove duplicates
-        filenames = {src.file: src.renamed_file(file_diff_dict)
-                     for src in result.affected_code}
-
-        editor_args = [editor] + list(filenames.values())
-        arg = EDITOR_ARGS.get(editor.strip(), None)
-        if arg:
-            editor_args.append(arg)
-
-        # Dear user, you wanted an editor, so you get it. But do you really
-        # think you can do better than we?
-        if editor in GUI_EDITORS:
-            subprocess.call(editor_args, stdout=subprocess.PIPE)
+        if editor in KNOWN_EDITORS:
+            editor_info = KNOWN_EDITORS[editor.strip()]
+        else:
+            # If the editor is unknown fall back to just passing
+            # the filenames
+            editor_info = {
+                'file_arg_template': '{filename}',
+                'gui': True
```

Comment:
My reasoning here was that the number of obscure gui editors is probably higher than that of obscure non-gui editors, so it's a better guess to go gui if we don't know it.